sympy/utilities/iterables.py

```diff
@@ -1381,6 +1381,12 @@ def partitions(n, m=None, k=None, size=False):
         raise ValueError("n must be >= 0")
     if m == 0:
         raise ValueError("m must be > 0")
+    if n == 0:
+        if size:
+            yield 1, {0: 1}
```

Comment:
I would feel better about not even allowing partitions of zero. What do you think about changing the error above to be "n must be > 0"?